jstests/aggregation/bugs/server33149.js

```diff
@@ -0,0 +1,20 @@
+(function() {
+    "use strict";
+
+    // For arrayEq.
+    load("jstests/aggregation/extras/utils.js");
+
+    let indexDB = db.getSiblingDB("create_indexess_db");
+    assert.commandWorked(indexDB.dropDatabase());
+
+    // Create an index with key: {x : 1}, name : "x_1
+    assert.commandWorked(indexDB.runCommand({ createIndexes : "test", indexes : [ { key : { x : 1 }, name : "x_1" } ] }));
+
+    // Create a duplicate index will not result an error
+    assert.commandWorked(indexDB.runCommand({ createIndexes : "test", indexes : [ { key : { x : 1 }, name : "x_1"} ] }), ErrorCodes.IndexAlreadyExists, "Identical index already exists ");
+
+    // Create an index that has same key pattern but with different name will fail
+    assert.commandFailedWithCode(indexDB.runCommand({ createIndexes : "test", indexes : [ { key : { x : 1 }, name : "x_2"} ] }), ErrorCodes.CannotCreateIndex, "Index with the same key pattern already exists with a different name");
```

Comment:
The third argument here is what will be printed if the assertion fails. I'd say you could leave it off, or say something like "expected not to be able to create index with different name and same options"